spec/factories/expertises.rb

```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :expertise do
```

Comment:
getting `FactoryBot::DuplicateDefinitionError:
  Factory already registered: expertise`
  error

Reply:
It seems that there's no such error from my side, did you come across this error when you run 'rspec'? For FactoryBot Expertise, it should have been defined once only in my latest code, so I am not sure why as well without having more context

Reply:
Yes i got this error by running ` rspec spec/factories/expertises.rb`

It seems like i already defined a factory for expertise elsewhere but i have no clue where it could be!

Reply:
Yes i got this error by running  rspec spec/factories/expertises.rb

It seems like i already defined a factory for expertise elsewhere but i have no clue where it could be!

Reply:
Same case for me if rspec spec/factories/expertises.rb. I tried to fix it but no luck. Below are my findings for reference:

- we only defined 'expertise' once for FB
- no issue with defining the FB configuration in rails helper
- if we add puts 'check', it will print two times, i.e. loading two times totally (and hence the duplication error)
- if we change 'factories' directory to anything else, the single factory file will not show error, but rspec (as a whole) will not be able to function (as FB's definition_file_paths setting points at directories which should be 'factories' by default)

I'm guessing the issue lies probably with the re-loading of the expertise factory at some point, which I am not sure if that's some configuration behind FB where we need to explicitly stop 're-loading' or 'automatic loading'

Since rspec overall looks fine, I suggest we seek review from Matt and ask him on this issue in parallel - see if he had encountered the same issue previously

Reply:
> Same case for me if rspec spec/factories/expertises.rb. I tried to fix it but no luck. Below are my findings for reference:
> 
> * we only defined 'expertise' once for FB
> * no issue with defining the FB configuration in rails helper
> * if we add puts 'check', it will print two times, i.e. loading two times totally (and hence the duplication error)
> * if we change 'factories' directory to anything else, the single factory file will not show error, but rspec (as a whole) will not be able to function (as FB's definition_file_paths setting points at directories which should be 'factories' by default)
> 
> I'm guessing the issue lies probably with the re-loading of the expertise factory at some point, which I am not sure if that's some configuration behind FB where we need to explicitly stop 're-loading' or 'automatic loading'
> 
> Since rspec overall looks fine, I suggest we seek review from Matt and ask him on this issue in parallel - see if he had encountered the same issue previously

Sorry, have been following this conversation, but didn't have the chance to reply until now. 

I believe the error is happening because FactoryBot already loads all factories by default at runtime, and then if you run the factory file itself, it tries to load that factory again, and throws the aforementioned error. Basically, as long as we don't run factory files, everything will be fine. But good investigation/discussion guys, always best to doublecheck instead of assuming that something is actually fine.